backend/assignments/1/src/main/java/org/example/CsvLoader.java

```diff
@@ -0,0 +1,95 @@
+package org.example;
+
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvException;
+
+import java.io.InputStreamReader;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility class to load CSV data.
+ */
+public class CsvLoader {
+
+    /**
+     * Load a list of coins from a CSV file.
+     *
+     * @param filePath The path to the CSV file containing coin data.
+     * @return A list of Coin objects.
+     */
+    public static List<Coin> loadCoins(String filePath) {
```

Comment:
More modularity expected.
Move utils, core logic etc in seperate classes/packages